src/xb-query.c

```diff
@@ -486,7 +507,7 @@ xb_query_new_full (XbSilo *silo, const gchar *xpath, XbQueryFlags flags, GError
 	g_return_val_if_fail (error == NULL || *error == NULL, NULL);
 
 	/* create */
-	priv->silo = g_object_ref (silo);
+	priv->silo = silo;  /* don’t take a reference, otherwise we get refcount loops with cached queries from xb_silo_lookup_query() */
```

Comment:
should we set a weak pointer thing?

Reply:
The `->silo` member is actually only ever used during parsing. I’ll split it out into a temporary parsing context.

Reply:
Yes, I think that's best, thanks.